rtrlib/lib/alloc_utils.h

```diff
@@ -34,4 +34,13 @@ void lrtr_free(void *ptr);
 
 void *lrtr_realloc(void *ptr, size_t size);
 
+/**
+ * @brief Duplicates a string
+ * @param[in] string
+ * @returns Duplicated string
+ * @returns NULL if string is NULL
+ * @returns NULL on error
```

Comment:
You are right.
I changed to function to match std strdup and added checks where necessary.